### ISSUE TITLE
Why do you need vue specific code?

### ISSUE THREAD

**Issue by stereobooster** [eldarc/vue-prerender#3]
Hey,

I was wondering why do you need vue specific code? You have full browser, you can crawl whole website without any specific code. Is there any reason to use `window._vuePrerender.$router.options.routes`, which I'm not aware of?

**Comment by eldarc** [comment]
@stereobooster You have three options:

1. Crawl the app, but if you are using the router programmatically, that link will be not crawled. For example if you use `$router.push('car-model/' + this.carModel) inside a component which has a a select tag:

```
<select v-model="carModel">
  <option v-for="_carModel in carModels" :value="carModel">{{ _carModel }}</option>
</select>
```
```
watch: {
    'carModel' (newValue) {
    this.$router.go('car-model/' + newValue)
  }
}
```

This is just an example, I'm using something for language switching.

2. Set a collection of predefined URLs, and those URLs will be rendered.

3. (This option is why I decided to use `window._vuePrerender.$router.options.routes`) Generate paths based on the router, and have an option to inject parameter values (this is an improved feature, I'm going to push today).

Extensive example:

You have following `vue-prerender` and router options:
```
{
  logLevel: 3,
  parseRouter: true,
  tidy: true,
  routerParams: {
    brand: {
      _default: ['', 'toyota', 'vw', 'opel', 'chevrolet'],
        model: {
        'unknown': ['other']
      }
    },
    model: {
      _default: ['unknown'],
        brand: {
        'toyota': ['corolla', 'yaris'],
          'vw': ['golf', 'polo'],
          'opel': ['astra', 'corsa'],
          'chevrolet': ['astra']
      }
    },
    year: {
      _default: [1950, 1951],
        model: {
        'yaris': [2014, 2013]
      },
      brand: {
        'other': [1840, 1841]
      }
    }
  }
}
```

```
export default new Router({
  mode: 'history',
  routes: [
    {
      path: '/',
      name: 'HelloWorld',
      component: HelloWorld
    },
    {
      path: '/cars/:brand?',
      // This will be rendered as `/cars/index.html` if `brand` isn't defined, because `brand` is optional.
      // However if `brand` is defined it will be rendered as `/cars/brand-name/index.html`
      // In the `vue-prerender` config when defining data sets for parameters, a blank string can be also included to generate
      // the route without the parameter.
      component: Prerender
    },
    {
      path: '/car-brand-by-year/:year',
      // If `year` is not defined it will be not rendered, because the `year` parameter is not optional.
      // The `year` parameter in `routerParams` needs to have a default array for it to have rendered paths.
      component: Prerender
    },
    {
      path: '/car/:brand/:model',
      // This will be rendered as `/car/brand-name/model-value/index.html'.
      // If any of the parameters in the configuration data sets for params isn't set, the route will be skipped,
      // because all params are defined as required.
      component: Prerender
    },
    {
      path: '/relations/:model/:brand',
      // This will be rendered as `/relations/model-value/brand-name/index.html'.
      // The params are the same as above, but the order is reversed.
      component: Prerender
    },
    {
      path: '/model/:model',
      // This will be rendered as `/model/model-value/index.html'.
      component: Prerender
    },
    {
      path: '/this-app/info/about-us/',
      // There are no params in this path, so it will be rendered just to `/this-app/info/about-us/index.html`
      component: Prerender
    }
  ]
})
```

This will generate a file like: `vw/golf/index.html`, but will not `toyota/golf/index.html`

With this you can get all the data you need and assign them to the configuration before prerendering.

In my particular case, I needed to inject all languages for each URL, like for example `/:_locale/about-us`, and locales array `['en', 'de', 'ja', ...]`. And later I'll have `publisher pages`, which I can get from the database before rendering.

**Comment by eldarc** [comment]
Related: #4

**Comment by stereobooster** [comment]
I see, but this also means that pages hidden behind `<select>` won't be discovered by google (if that matters). You can use custom dropdown (`<div>`) with links inside.

I chose crawling strategy in [my project](https://github.com/stereobooster/react-snap), that is why I asked. I was wondering what I missed. Thanks

**Comment by stereobooster** [comment]
Also you can list language root pages as starting point and the crawling, like

```
pages: ["/en", "/de" ...]
```

And later crawler will discover everything else

```
/en -> /en/a, /en/b
/de -> /de/a, /de/b
...
```

**Comment by eldarc** [comment]
Well, if it's hidden as a link, it doesn't mean it can't be in the sitemap.xml (maybe I'll add this feature at a later stage) and indexed that way, or shared on social media. But yeah, this can be solved with hidden links, but requires manual maintenance. If the user of the plugin prefers this, it's there and can be used.

However, If I have an item hidden from the starting filter view. For example `/item/:id` (`/item/9991`), and isn't filtered in the `/items` path, I either need to have a way of paging, or generate a hidden list of all items. So I decided that implementing parameters injection is easier. If I don't have pre-renedring for all public items, a share action on Facebook, Twitter or any other social media could result that maybe the item's custom meta description isn't parsed and displayed there.

But I've also got another use-case. I'm building up another plugin for Vue called `vue-i18n-gettext`, which will have automatic prefixing for translated URLs, and also translating parameters inside the translated URL. I'm not 100% sure if everything will go together as planned, but that's the idea. So the list of languages can be loaded from the translation plugin, and If I change the locales (add or remove some), I don't have to think about maintaining the list of hidden URLs, or changing a lot of the configuration (it can be central). But also, the plugin will have to possibility that the translated URLs can have different formats, like `/blog/:_locale-hello`, so it can be `/blog/de-hello`.

I'm trying to assume as little as possible, and make it flexible. At the end the result could be that I complicated too much, but we'll see. 😄 

@stereobooster Thank you for your feedback!
